plugins/src/main/java/org/tron/plugins/DbLite.java

```diff
@@ -585,16 +570,30 @@ private void mergeBak2Database(String liteDir, BlockNumInfo blockNumInfo) throws
 
   private byte[] getDataFromSourceDB(String sourceDir, String dbName, byte[] key)
           throws IOException, RocksDBException {
+    byte[] keyInCp = Bytes.concat(simpleEncode(dbName), key);
+    byte[] valueInCp = null;
     DBInterface sourceDb = DbTool.getDB(sourceDir, dbName);
-    DBInterface checkpointDb = getCheckpointDb(sourceDir);
-    // get data from tmp first.
-    byte[] valueFromTmp = checkpointDb.get(Bytes.concat(simpleEncode(dbName), key));
+    // get data from checkpoint first.
+    List<String> cpList = getCheckpointV2List(sourceDir);
+    if (cpList.size() > 0) {
+      // reverse iteration
+      Collections.reverse(cpList);
+      for (String cp: cpList) {
```

Comment:
It may be a bit slow because it is within a two-level loop, and merging these checkpoints into one database before all other operations may be more efficient.